### PR TITLE
Compress context and fetch endpoints in initial state

### PR DESCRIPTION
**Pull request by bitterpanda63**

### CODE REVIEW

File: aikido_firewall/sources/functions/request_handler.py [hunk 1]
```diff
@@ -1,15 +1,26 @@
...
+import threading
 from aikido_firewall.background_process import get_comms
 from aikido_firewall.context import get_current_context
 from aikido_firewall.helpers.is_useful_route import is_useful_route
+from aikido_firewall.helpers.logging import logger
+from aikido_firewall.background_process.ipc_lifecycle_cache import (
+    IPCLifecycleCache,
+)
+
+local = threading.local()
```
Comment (willem-delbare) [inline aikido_firewall/sources/functions/request_handler.py]: can you comment what the goal is?

File: aikido_firewall/vulnerabilities/__init__.py [hunk 2]
```diff
@@ -24,6 +25,9 @@
 from .path_traversal.check_context_for_path_traversal import (
     check_context_for_path_traversal,
 )
+from aikido_firewall.background_process.ipc_lifecycle_cache import get_cache
+
+local = threading.local()
```
Comment (willem-delbare) [inline aikido_firewall/vulnerabilities/__init__.py]: comment about the goal

File: aikido_firewall/background_process/commands/get_bypassed_ips.py [hunk 3]
```diff
@@ -0,0 +1,7 @@
+"""Mainly exports `process_get_bypassed_ips`"""
+
+
+def process_get_bypassed_ips(bg_process, data, conn):
+    """Checks if the ip is on the bypass list"""
+    bypassed_ips = bg_process.reporter.conf.bypassed_ips
+    conn.send(bypassed_ips)
```
Comment (willem-delbare) [inline aikido_firewall/background_process/commands/get_bypassed_ips.py]: let's add some comments here... function is called 'get' but it does not return anything.. and rather it 'sends'?

File: aikido_firewall/sources/functions/request_handler.py (hunk 1 above)
Comment (bitterpanda63) [reply aikido_firewall/sources/functions/request_handler.py]: Woops that should've been rm'ed

File: aikido_firewall/vulnerabilities/__init__.py (hunk 2 above)
Comment (bitterpanda63) [reply aikido_firewall/vulnerabilities/__init__.py]: Same here, this logic moved to the ipc_lifecycle_cache file

File: aikido_firewall/background_process/commands/get_bypassed_ips.py (hunk 3 above)
Comment (bitterpanda63) [reply aikido_firewall/background_process/commands/get_bypassed_ips.py]: Dis better?

File: aikido_firewall/background_process/ipc_lifecycle_cache.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
+"""
+Contains the `IPCLifecycleCache` cache
```
Comment (willem-delbare) [inline aikido_firewall/background_process/ipc_lifecycle_cache.py]:
```suggestion
Contains the `IPCLifecycleCache` cache for the duration of a single request
```

File: aikido_firewall/background_process/ipc_lifecycle_cache.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
+"""
+Contains the `IPCLifecycleCache` cache for the duration of a single request
+"""
+
+import threading
+from .comms import get_comms
+
+local = threading.local()
```
Comment (willem-delbare) [inline aikido_firewall/background_process/ipc_lifecycle_cache.py:9]:
```suggestion
local = threading.local() #  cache needs to be per thread, not shared in the entire process
```

**Pull request merged**